### PR TITLE
nixos/gitea: fix mysql issue, add mysql socket auth, and add a nixos test

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
My gitea instance has been crashing every few nights because the systemd service is assuming a postgresql database.

###### Things done
- fixed the issue causing my gitea instance to crash on start sometimes because it is using a mysql database
- added mysql socket authentication
- added a nixos test for gitea

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

ping @disassembler @etu

### CODE REVIEW

**Comment by aanderse** [comment]
Note failure on aarch64-linux isn't unexpected... https://github.com/NixOS/nixpkgs/pull/45968

Anyone know why that isn't unexpected? Can we fix it?

@GrahamcOfBorg test gitea

**Comment by etu** [comment]
@aanderse I was thinking, not related to the aarch64-failure. But maybe we should have a test machine in the same test file for postgres as well? Just to make sure that both works?

**Comment by aanderse** [comment]
Still working on postgres test as it errors out in the pre-start script. More to come.

**Comment by aanderse** [comment]
@GrahamcOfBorg test postgresql

Something funny going on with my machine...

**Comment by aanderse** [comment]
@etu if you're happy with this I'll look for someone to merge.

File: nixos/modules/services/misc/gitea.nix [hunk 1]
```diff
@@ -253,7 +261,7 @@ in
 
     systemd.services.gitea = {
       description = "gitea";
-      after = [ "network.target" "postgresql.service" ];
+      after = [ "network.target" "postgresql.service" ] ++ lib.optional usePostgresql "postgresql.service" ++ lib.optional useMysql "mysql.service";
```
[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/misc/gitea.nix]: Thanks for catching!

**Pull request merged**